labonneboite/common/load_data.py

```diff
@@ -28,35 +28,69 @@ def f(full_filename):
 
     for row in reader:
         if len_previous_row:
+            # at least second line of CSV file
             if len(row) != len_previous_row:
                 raise Exception("found rows with different number of fields : %s" % row)
-        rows.append(row)
+            rows.append(row)
+        else:
+            # first line of CSV file: headers should be ignored
+            pass
+        
         len_previous_row = len(row)
 
     return rows
 
 
+def load_rows_as_dict(rows):
+    d = {}
+    for row in rows:
+        if len(row) != 2:
+            raise Exception("wrong number of fields")
```

Comment:
Est-ce qu'on peut lever autre chose qu'une `Exception`, pour pouvoir éventuellement l'attraper plus tard ? `ValueError` ou `IndexError` par exemple ?

Reply:
ok pour une `IndexError` cf

https://docs.python.org/2/library/exceptions.html
Raised when a sequence subscript is out of range.